oauth2_client/providers/salesforce/provider.py

```diff
@@ -0,0 +1,151 @@
+"""
+SalesforceProvider.
+Currently supports only JWT Token Bearer OAuth2 flow.
+"""
+import json
+from base64 import urlsafe_b64encode
+from datetime import datetime, timedelta
+
+import requests
+from Crypto.Hash import SHA256
+from Crypto.PublicKey import RSA
+from Crypto.Signature import PKCS1_v1_5
+
+from labsterutils.obj import get_value_or_default
+from oauth2_client.models import AccessToken
+from oauth2_client.providers.oauth2.provider import OAuth2Provider
+from oauth2_client.compat import timestamp
+
+
+class SalesforceProvider(OAuth2Provider):
+    id = 'salesforce'
+    name = 'Salesforce'
+
+    JWT_HEADER = '{"alg":"RS256"}'
+    GRANT_TYPE = "urn:ietf:params:oauth:grant-type:jwt-bearer"
+    AUDIENCE_URI = 'https://login.salesforce.com'
+    SANDBOX_AUDIENCE_URI = 'https://test.salesforce.com'
+    AUTH_URI = 'https://login.salesforce.com/services/oauth2/token'
+    SANDBOX_AUTH_URI = 'https://test.salesforce.com/services/oauth2/token'
+
+    def fetch_token_jwt_bearer(self):
```

Comment:
is there a way to use `authorization_code` grant type instead of `jwt_bearer`? Allauth implements `authorization_code` flow if I'm not mistaken.

`urn:ietf:params:oauth:grant-type:jwt-bearer` doesn't seem to be Salesforce custom implementation (https://tools.ietf.org/html/rfc7523), so does it make sense to move this logic to Oauth2 provider?

Reply:
not relevant anymore after simplification, marking this as resolved